### PR TITLE
Feature/as a user i want to add my transaction to my wallet

### PR DESCRIPTION
**Pull request by MackowiakPiotr**

### CODE REVIEW

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionController.java [hunk 1]
```diff
@@ -3,26 +3,33 @@
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
```
Comment (odzio33) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionController.java:7]: the wildcard import approach has advantages, but its usage can cause problems in certain situations

please turn off Wildcard imports (https://intellij-support.jetbrains.com/hc/en-us/community/posts/206203659-Turn-off-Wildcard-imports-)

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionController.java [hunk 2]
```diff
@@ -3,26 +3,33 @@
...
 import pl.byczazagroda.trackexpensesappbackend.service.FinancialTransactionService;
 
+import javax.validation.Valid;
 import javax.validation.constraints.Min;
 import javax.validation.constraints.NotNull;
 
+@Validated
 @RestController
 @RequiredArgsConstructor
 @RequestMapping("api/transactions/")
```
Comment (odzio33) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/controller/FinancialTransactionController.java:19]: please remove last backslash from @RequestMapping("api/transactions/")

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/CreateFinancialTransactionDTO.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package pl.byczazagroda.trackexpensesappbackend.dto;
+
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+
+import javax.validation.constraints.DecimalMin;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import java.math.BigDecimal;
+
+public record CreateFinancialTransactionDTO(@Min(1) @NotNull Long walletId, @DecimalMin("0.0") BigDecimal amount,
```
Comment (odzio33) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/CreateFinancialTransactionDTO.java:10]: Please refactor annotation `@DecimalMin("0.0")` to `@DecimalMin("0")`
Please add annotation `@Digits("13,2")`

**Review by odzio33 — CHANGES_REQUESTED** [review]
changes required

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java [hunk 4]
```diff
@@ -20,6 +25,26 @@ public class FinancialTransactionServiceImpl implements FinancialTransactionServ
...
+            throw new AppRuntimeException(
+                    ErrorCode.W003,
+                    String.format("Wallet with id: %d does not exist", walletId));
+        });
+        FinancialTransaction financialTransaction = new FinancialTransaction();
+        financialTransaction.setTransactionDate(Instant.now());
+        financialTransaction.setAmount(createFinancialTransactionDTO.amount());
+        financialTransaction.setWallet(wallet);
+        financialTransaction.setFinancialTransactionType(createFinancialTransactionDTO.financialTransactionType());
+        financialTransaction.setDescription(createFinancialTransactionDTO.description());
```
Comment (DawidStuzynski) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java:43]: I think that using @builder would be better

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/CreateFinancialTransactionDTO.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+
+import javax.validation.constraints.Digits;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.PositiveOrZero;
+import java.math.BigDecimal;
+
+public record CreateFinancialTransactionDTO(@Min(1) @NotNull Long walletId, @Digits(integer = 13, fraction = 2) @PositiveOrZero BigDecimal amount,
+                                            String description, FinancialTransactionType financialTransactionType) {
```
Comment (odzio33) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/dto/CreateFinancialTransactionDTO.java:12]: Please add validation for Enum (FinancialTransactionType ) type

**Review by odzio33 — CHANGES_REQUESTED** [review]
Changes required

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Review by dawciobiel — APPROVED** [review]
Looks good for me

**Review by dawciobiel — APPROVED** [review]
Looks goods for me

**Review by odzio33 — APPROVED** [review]
Looks good for me

**Pull request merged**